Tests/MapboxNavigationTests/VanishingRouteLineTests.swift

```diff
@@ -107,36 +128,78 @@ class VanishingRouteLineTests: TestCase {
+        XCTAssertEqual(routePoints.flatList.count, 6)
+        // This route has 3 steps.
+        XCTAssertEqual(routePoints.nestedList.flatMap{$0}.count, 3)
+        
+        guard let routeLineGranularDistances = navigationMapView.calculateGranularDistances(routePoints.flatList) else {
+            XCTFail("Failed to calculate granular distances.")
+            return
+        }
+        let distanceArray = routeLineGranularDistances.distanceArray
+        XCTAssertEqual(distanceArray.count, routePoints.flatList.count)
+        XCTAssertEqual(distanceArray.first?.distanceRemaining, routeLineGranularDistances.distance)
+        XCTAssertEqual(distanceArray.first?.point, routePoints.flatList.first)
+        XCTAssertEqual(distanceArray.last?.distanceRemaining, 0.0)
+        XCTAssertEqual(distanceArray.last?.point, routePoints.flatList.last)
     }
     
     func testUpdateUpcomingRoutePointIndex() {
         // https://github.com/mapbox/mapbox-navigation-android/blob/0ca183f7cb7bec930521ea9bcd59d0e8e2bef165/libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApiTest.kt#L802
         let route = getRoute()
+        let routeProgress = getRouteProgress()
         
         navigationMapView.initPrimaryRoutePoints(route: route)
-        navigationMapView.routeLineGranularDistances = nil
-        XCTAssertEqual(navigationMapView.fractionTraveled, 0.0)
-        
-        let routeProgress = getRouteProgress()
+        guard let flatList = navigationMapView.routePoints?.flatList,
+              let nestedPoints = navigationMapView.routePoints?.nestedList,
+              let currentLegPoints = nestedPoints.first else {
+            XCTFail("Failed to initialize primary route points.")
+            return
+        }
+        // This route has 9 coordinates.
+        XCTAssertEqual(flatList.count, 9)
+        // This route has 1 leg.
+        XCTAssertEqual(nestedPoints.count, 1)
+        // This route leg has 4 steps.
+        XCTAssertEqual(currentLegPoints.count, 4)
+        XCTAssertEqual(currentLegPoints[safe: 2]?.count, routeProgress.currentLegProgress.currentStep.shape?.coordinates.count)
         
         navigationMapView.updateUpcomingRoutePointIndex(routeProgress: routeProgress)
+        XCTAssertEqual(routeProgress.currentLegProgress.currentStepProgress.distanceTraveled, 0.0)
         
-        // Because mapbox-directions-swift parses the route with one more duplicate coordinate in the last step of each route leg.
-        // The one leg route has one more coordinate compared with Android.
-        XCTAssertEqual(navigationMapView.routeRemainingDistancesIndex, 7)
+        var traveledPoints = 0
+        traveledPoints += currentLegPoints[0].count
+        traveledPoints += currentLegPoints[1].count
+        // Because user has arrived at the first coordinate of current step with distanceTraveled on this step as 0.0.
+        traveledPoints += 1
```

Comment:
Why not make it a one liner constant?

Reply:
Thanks for the suggestion, the constant is applied instead of the steps calculation.